Orange/widgets/data/owfile.py

```diff
@@ -142,6 +146,142 @@ def __repr__(self):
     __str__ = __repr__
 
 
+class VarTableModel(QtCore.QAbstractTableModel):
+    places = "feature", "class", "meta", "skip"
+    typenames = "nominal", "numeric", "string"
+    vartypes = DiscreteVariable, ContinuousVariable, StringVariable
+    name2type = dict(zip(typenames, vartypes))
+    type2name = dict(zip(vartypes, typenames))
+
+    def __init__(self, widget):
+        super().__init__()
+        self.widget = widget
+        self.variables = []
+
+    def set_domain(self, domain):
+        def may_be_numeric(var):
+            if var.is_continuous:
+                return True
+            if var.is_discrete:
+                try:
+                    [float(x) for x in var.values]
```

Comment:
It may not matter much in this case, since `len(var.values)` is usually small, but in general, always prefer avoiding building unnecessary intermediary lists, i.e., `sum(float(x) for x in var.values)`.